app/routes/image-set.js

```diff
@@ -3,6 +3,17 @@ import OrganizationRouteMixin from 'lion-guardians/mixins/organization-route';
 import RequireUserMixin from 'lion-guardians/mixins/require-user';
 
 export default Ember.Route.extend( OrganizationRouteMixin, RequireUserMixin, {
+  beforeModel: function() {
+    var route = this;
+    return this._super().then(function() {
+      if (window.SKIP_MAPS) {
+        return Ember.RSVP.resolve();
+      } else {
+        return route.loadGoogleMap();
```

Comment:
You could likely use an initializer to inject it onto components, or your component could import the [lazy loading function](https://github.com/huafu/ember-google-map/blob/6257f5febe1bfa0a6776c7ffd07e4fcc9e13258f/addon/utils/load-google-map.js) and do it itself.

Here's the addon's initializer: https://github.com/huafu/ember-google-map/blob/6257f5febe1bfa0a6776c7ffd07e4fcc9e13258f/app/initializers/ember-google-map.js
You could make your own that did roughly the same thing. The import path would be the same.

There's nothing stopping you from putting the `loadGoogleMap` function into the controller's scope, from which it could be passed into the component, too:
```
// route
this.controller.set('loadGoogleMap', this.loadGoogleMap);

// template
{{lg-google-map loadGoogleMap=loadGoogleMap}}

// in compnent
var loadGoogleMap = this.get('loadGoogleMap');
loadGoogleMap().then(function(){
  // loaded
});
```

Injecting it onto components yourself or importing the function in the component (`import loadGoogleMap from 'ember-google-map/utils/load-google-map';`) are better ways, though.

However, I think there is value in displaying the map even when not editing, because a user may want to edit the image set because the location is incorrect. If the location/map is not displayed by default, though, they wouldn't know that it is correct and wouldn't want to edit it. (The [google maps static image api](https://developers.google.com/maps/documentation/imageapis/) may be more appropriate for non-interactive map displays, though.)

Thanks for being cognizant of the bandwidth/loading considerations and addressing them here, though.